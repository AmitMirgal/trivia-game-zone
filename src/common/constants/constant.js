export const HOME_TITLE = "Welcome to the Trivia Challenge!";

export const HOME_DESCRIPTION =
  "You will be presented with 10 True or False question. can you score 100%?";
