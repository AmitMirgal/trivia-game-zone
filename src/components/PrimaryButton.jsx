import React from "react";
import styled from "styled-components";
import { PropTypes } from "prop-types";
import { useWhyDidYouUpdate } from "../pages/useWhyDidYouUpdate";

const Button = styled.button`
  background: ${(props) => (props.primary ? "#005dff" : "white")};
  color: ${(props) => (props.primary ? "white" : "#005dff")};
  font-size: 18px;
  font-weight: 600;
  margin: 8px;
  padding: 0 8px;
  border: 2px solid #005dff;
  border-radius: 30px;
  height: 45px;
  width: 200px;
  cursor: pointer;
  outline: none;

  &:hover {
    background: ${(props) => (props.primary ? "#002f80" : "white")};
    border: 2px solid #002f80;
  }
`;

const PrimaryButton = (props) => {
  useWhyDidYouUpdate("primary button", props);

  const { label, primary, onClick } = props;

  return (
    <Button primary={primary} onClick={onClick}>
      {label}
    </Button>
  );
};

PrimaryButton.propTypes = {
  label: PropTypes.node,
  primary: PropTypes.bool,
  onClick: PropTypes.func,
};

export default PrimaryButton;
