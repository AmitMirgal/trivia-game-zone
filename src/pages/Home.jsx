import React from "react";
import PrimaryButton from "../components/PrimaryButton";

import { ReactComponent as Arrow } from "../common/assets/arrow.svg";

import styled from "styled-components";
import { useHistory } from "react-router-dom";

import { HOME_TITLE, HOME_DESCRIPTION } from "../common/constants/constant";

const HomePanel = styled.div`
  display: flex;
  flex-direction: column;
  font-size: 14px;
  color: #005dff;
  align-items: flex-start;
  padding: 16px;
  margin: 16px;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`;

const HomeTitle = styled.h1`
  font-size: 60px;
  font-weight: 700;
  color: #2d3748;
`;

const HomeDescription = styled.p`
  font-size: 18px;
  font-weight: 400;
  color: #a0aec0;
`;

const Home = () => {
  const history = useHistory();

  const beginTest = () => {
    history.push("/quiz");
  };

  return (
    <HomePanel>
      <HomeTitle>{HOME_TITLE}</HomeTitle>
      <HomeDescription>{HOME_DESCRIPTION}</HomeDescription>

      <Arrow />

      <PrimaryButton label="Begin" primary onClick={beginTest} />
    </HomePanel>
  );
};

export default Home;
