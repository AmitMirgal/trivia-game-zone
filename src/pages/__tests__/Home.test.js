import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import Home from "../Home";
import { HOME_TITLE, HOME_DESCRIPTION } from "../../common/constants/constant";
import PrimaryButton from "../../components/PrimaryButton";

describe("<Home/>", () => {
  let wrapper;
  const prepare = () => {
    wrapper = mount(<Home />);
  };

  it("renders Home component", () => {
    // when
    prepare();

    // then
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Home title", () => {
    // when
    prepare();

    // then
    expect(wrapper.find("h1").text()).toBe(HOME_TITLE);
  });

  it("Home description", () => {
    // when
    prepare();

    // then
    expect(wrapper.find("p").text()).toBe(HOME_DESCRIPTION);
  });

  it("button label is Begin", () => {
    // when
    prepare();

    // then
    expect(wrapper.find("button").text()).toBe("Begin");
  });

  it("button is triggered", () => {
    // when
    const mockCallBack = jest.fn();
    const buttonWrapper = mount(
      <PrimaryButton onClick={mockCallBack}>Begin</PrimaryButton>
    );

    buttonWrapper.find("button").simulate("click");

    // then
    expect(mockCallBack).toHaveBeenCalled();
  });
});
