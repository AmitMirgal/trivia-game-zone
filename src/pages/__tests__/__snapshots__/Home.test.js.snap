// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> renders Home component 1`] = `
<Home>
  <styled.div>
    <div
      className="sc-AxirZ MSAgO"
    >
      <styled.h1>
        <h1
          className="sc-AxiKw fTyEPL"
        >
          Welcome to the Trivia Challenge!
        </h1>
      </styled.h1>
      <styled.p>
        <p
          className="sc-AxhCb egbyGX"
        >
          You will be presented with 10 True or False question. can you score 100%?
        </p>
      </styled.p>
      <ForwardRef(SvgArrow)>
        <svg>
          arrow.svg
        </svg>
      </ForwardRef(SvgArrow)>
      <PrimaryButton
        label="Begin"
        onClick={[Function]}
        primary={true}
      >
        <styled.button
          onClick={[Function]}
          primary={true}
        >
          <button
            className="sc-AxjAm dtKtzu"
            onClick={[Function]}
          >
            Begin
          </button>
        </styled.button>
      </PrimaryButton>
    </div>
  </styled.div>
</Home>
`;
